src/AnalysisEpic.cxx

```diff
@@ -268,6 +268,26 @@ void AnalysisEpic::Execute()
 	}
 	// Add the final state particle to the HFS
 	kin->AddToHFS(recpart_.vecPart);
+
+	// Add reconstructed particle and true info to HFSTree
+	if( writeHFSTree ){
+	  int pid = recpart_.pid;
+	  int truepid;
+	  int mcid_ = recpart_.mcID;
+	  TLorentzVector truep4;
+	  if(mcid_ >= 0) {
+	    for(auto imc : mcpart) {
+	      if(mcid_ == imc.mcID) {
+		truep4 = imc.vecPart;
+		truepid = imc.pid;
+		break;
+	      }
+	    }
+	  }	  
+	  kin->AddToHFSTree(recpart_.vecPart, pid,
+			    truep4, truepid
+			    );
```

Comment:
should this be inside the `if(mcid_ >= 0)` block, or in the inner `if(mcid_==imc.mcID)` block?